tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -1113,13 +1118,35 @@ def _run_pseudolabeling(self, unlabeled_data: pd.DataFrame, max_iter: int,
                 X_test = X_test.loc[test_pseudo_idxes[~test_pseudo_idxes].index]
                 previous_score = current_score
 
+        # Should be done at the end. When done at each iteration caused the model to get much worse
+        if fit_ensemble:
+            logger.log(15, 'Fitting weighted ensemble using models trained with pseudo labeled data')
+
+            if return_pred_prob:
+                logger.log(30, "WARNING: 'return_pred_prob' is set to True and 'fit_ensemble' is set to true."
+                               " Fit ensemble is done at the end of training so predictions returned are not"
+                               " done by the model ensembling models trained using pseudo labeled data.")
```

Comment:
this statement does not make sense

Reply:
also in this situation, why can't we just return pred_probas from the final ensemble?

Reply:
return pred probs from final model